libdoipserver/src/VehicleIdentificationHandler.cpp

```diff
@@ -1,20 +1,23 @@
 #include "VehicleIdentificationHandler.h"
 #include <iostream>
 
-
-
 unsigned char* createVehicleIdentificationResponse(std::string VIN,unsigned short LogicalAddress, 
                                                     unsigned char* EID, unsigned char* GID,
                                                     unsigned char FurtherActionReq) //also used für the Vehicle Announcement
 {
     unsigned char* message = createGenericHeader(PayloadType::VEHICLEIDENTRESPONSE, _VIResponseLength);
     
     //VIN Number 
-    int j = 0;
+    unsigned int j = 0;
     for(int i = 8; i <= 24; i++)
-    {      
-        message[i] = (unsigned char)VIN[j];
-        j++;
+    {
+        if (j < VIN.length()){
+            message[i] = (unsigned char)VIN[j];
+            j++;
+	} else {
+	    //Pad with zero if VIN is shorter than 16 bytes
+	    message[i] = 0;
```

Comment:
Not sure if it makes sense to pad with 0.
DoIP does not define the VIN as a Zero-Terminated string, so a client will likely use all 17 (not 16 as written in the comment!) bytes as chars.
I would suggest to either terminate the whole process on startup of the server when the length of the configured VIN is not 17.
Or pad with one of the characters allowed for a VIN (German wikipedia has a nice list). I suggest 'X' or '0'.

Reply:
I figured, if the VIN is ASCII, a padding with \0 would make it easier to parse it as string. But other symbols would be fine for me, too.
Oh, and thanks for pointing out the 17 bytes.

If we terminate the process on startup in case of an invalid VIN format, then the example code needs to be fixed, too. As that was referenced in the bug report.

Reply:
This code prepares the VIN for transmitting. ISO13400-2 defines a fixed 17 bytes length for the VIN and no zero termination. Therefore we should not zero-terminate the transmitted VIN but always send 17 valid characters.

Reply:
Updated the code to use '0' instead. And fixed the length naming.
Also added a comment to the example code.